docs/en/v0.4/user-guide/upgrade.md

```diff
@@ -0,0 +1,128 @@
+        --max-retry <MAX_RETRY>        Max retry times for each job [default: 3]
+        --output-dir <OUTPUT_DIR>      Directory to put the exported data. E.g.:
+                                       /tmp/greptimedb-export
+    -t, --target <TARGET>              Things to export [possible values: create-table, table-data]
+```
+
+Here explains the meaning of some important options
+
+- `--addr`: The address of the Frontend node.
+- `--output-dir`: The directory to put the exported data. Give a path at your current machine. The exported SQL files will be put in that directory.
+- `--target`: The things to export. `create-table` can export the `CREATE TABLE` clause for each table. `table-data` can export the data of each table alone with `COPY FROM` clause.
+
+For a complete upgrade, you will need to execute this tools twice with each target options.
+
+## Example
+
+Here is a complete example for upgrading from `v0.3.0` to `v0.4.0`.
+
+In the following text, we assume that you have a Frontend's gRPC endpoint is available at `127.0.0.1:4000`. And the catalog is `greptime`, schema is `public`. Hence the database is `greptime-public`. The output dir is `/tmp/greptimedb-export`.
+
+### Export `CREATE TABLE`
+
+```shell
+greptime cli export --addr '127.0.0.1:4001' --database greptime-public --output-dir /tmp/greptimedb-export --target create-table
+```
+
+If success, you will see something like this
+
+```log
+2023-10-20T09:41:06.500390Z  INFO cmd::cli::export: finished exporting greptime.public with 434 tables
+2023-10-20T09:41:06.500482Z  INFO cmd::cli::export: success 1/1 jobs
+```
+
+And now the output directory structure is
+
+```plaintext
+/tmp/greptimedb-export
+└── greptime-public.sql
+```
+
+### Export table data
+
+```shell
+greptime cli export --addr '127.0.0.1:4001' --database greptime-public --output-dir /tmp/greptimedb-export --target table-data
+```
+
+The log output is similar to the previous one. And the output directory structure is
+
+```plaintext
+/tmp/greptimedb-export
+├── greptime-public
+│   ├── up.parquet
+│   └── other-tables.parquet
+├── greptime-public_copy_from.sql
+└── greptime-public.sql
+```
+
+New files are `greptime-public_copy_from.sql` and `greptime-public`. The former one contains the `COPY FROM` clause for each table. The latter one contains the data of each table.
+
+### Import table schema and data
+
+Then you need to execute SQL files generated by the previous step. First is `greptime-public.sql`. Using `psql` as the client:
+
+:::tip NOTICE
+From this step, all the operation is done in the new version of GreptimeDB.
+:::
+
+Before executing the following command, you need first to create the corresponding database in your new deployment (but in this example, the database `greptime-public` is the default one).
+
+This command will create all the tables in the new version of GreptimeDB.
+
+```shell
+psql greptime-public -f /tmp/greptime-public.sql
```

Comment:
I believe the command should be `psql -h 127.0.0.1 -p 4003 -d public`. Just use `public` to specify schema would be enough since catalog is auto-completed to `greptime`